Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.py

```diff
@@ -324,38 +334,55 @@ def storage_blob_containers_list_request(self, args):
 
         return self.ms_client.http_request(
             method='GET',
-            url_suffix=url,
+            full_url=full_url,
             params=params,
         )
 
-    def storage_blob_containers_delete_request(self, args):
+    def storage_blob_containers_delete_request(self, subscription_id: str | None,
+                                               resource_group_name: str | None,
+                                               account_name: str | None, container_name: str | None) -> requests.Response:
 
         return self.ms_client.http_request(
             method='DELETE',
-            url_suffix=f'/resourceGroups/{self.resource_group_name}/providers/Microsoft.Storage/storageAccounts/'
-                       f'{args["account_name"]}/blobServices/default/containers/{args["container_name"]}',
-            params={
-                'api-version': API_VERSION,
-            },
+            full_url=f'https://management.azure.com/subscriptions/{subscription_id}/resourceGroups/{resource_group_name}/providers/Microsoft.Storage/\
+storageAccounts/{account_name}/blobServices/default/containers/{container_name}?api-version={API_VERSION}',
             resp_type='response'
         )
 
+    def storage_subscriptions_list_request(self):
+        return self.ms_client.http_request(
+            method='GET',
+            full_url=f'https://management.azure.com/subscriptions?api-version={API_VERSION}')
+
+    def list_resource_groups_request(self, subscription_id: str | None,
+                                     filter_by_tag: str | None, limit: str | int | None) -> Dict:
```

Comment:
here as well, it is a bit strange that it could be an int, str or None.

Reply:
It is the truth, all types are valid, but I will change it to avoid confusion.

Reply:
let's stick with int | None.

Reply:
fixed.